plugin/core/protocol.py

```diff
@@ -411,3 +411,6 @@ def to_lsp(self) -> Dict[str, str]:
 
     def uri(self) -> str:
         return filename_to_uri(self.path)
+
+    def includes_uri(self, uri: str) -> bool:
+        return uri.startswith(self.uri())
```

Comment:
Does this always work correctly on Windows?

Reply:
It looks like it will work. When converting from URI to file path the case is normalized (through the code used in `uri_to_filename`).
```
>>> import urllib
>>> urllib.request.url2pathname(urllib.parse.urlparse('file:///c:/Users/IEUser/LSP/plugin/core/url.py').path)
'C:\\Users\\IEUser\\LSP\\plugin\\core\\url.py'
>>> urllib.request.url2pathname(urllib.parse.urlparse('file:///C:/Users/IEUser/LSP/plugin/core/url.py').path)
'C:\\Users\\IEUser\\LSP\\plugin\\core\\url.py'

Reply:
I should have probably tested `filename_to_uri` in this case so here it is:
```
>>> urllib.request.pathname2url('C:\\Users\\IEUser\\LSP\\plugin\\core\\url.py')
'///C:/Users/IEUser/LSP/plugin/core/url.py'
>>> urllib.request.pathname2url('c:\\Users\\IEUser\\LSP\\plugin\\core\\url.py')
'///C:/Users/IEUser/LSP/plugin/core/url.py'
```
also works as expected regardless of the drive case.

And if you are thinking about the rest of the path besides just the drive letter then I guess that won't be normalized.

Reply:
But then I don't see how would it be possible for the case of the rest of the path to get out of sync with what we sent initially to the server. I understand that the server might normalize the drive letter's case but it shouldn't change anything else. What would be the point of that?